BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -139,6 +141,32 @@ public ICollection<ITelemetry> TelemetryItems
             get; private set;
         }
 
+        /// <summary>
+        /// Gets the flush async id for the transmission.
+        /// </summary>
+        /// [EditorBrowsable(EditorBrowsableState.Never)]
+        public long FlushAsyncId { get; } = Interlocked.Increment(ref flushAsyncCounter);
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this transmission is created from FlushAsync API call.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool HasFlushTask { get; set; } = false;
+
+        /// <summary>
+        /// Serializes telemetry items.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Serialize(Uri address, IEnumerable<ITelemetry> telemetryItems, TimeSpan timeout = default(TimeSpan))
```

Comment:
minor point. I could be wrong as well.

feels a bit awkward for a instance method Serialize accept a list of items to be serialized...
Could we store the items in the ctor() itself, and Serialize() call simply serializes the items it already has?

Reply:
This is tricky, as we already have a constructor which takes the same number of arguments and serializes it.